pyccl/ccl_massfunc.i

```diff
@@ -7,55 +7,61 @@
 %include "../include/ccl_massfunc.h"
 
 // Enable vectorised arguments for arrays
-%apply (double* IN_ARRAY1, int DIM1) {(double* halo_mass, int nm)};
-%apply (double* ARGOUT_ARRAY1, int DIM1) {(double* output, int nout)};
+%apply (double *IN_ARRAY1, int DIM1) {(double *halo_mass, int nm)};
+%apply (int DIM1, double *ARGOUT_ARRAY1) {(int nout, double *output)};
+
+/* The python code here will be executed before all of the functions that
+   follow this directive. */
+%feature("pythonprepend") %{
+    if numpy.shape(halo_mass) != (nout,):
+        raise CCLError("Input shape for `halo_mass` must match `(nout,)`!")
+%}
 
 %inline %{
-void massfunc_vec(ccl_cosmology * cosmo,
-                    double a, double odelta,
-                    double* halo_mass, int nm,
-                    double* output, int nout,
-                    int* status)
-{
-    assert(nout == nm);
-    for(int i=0; i < nm; i++){
+void massfunc_vec(
+        ccl_cosmology * cosmo,
+        double a, double odelta,
+        double *halo_mass, int nm,
+        int nout, double *output,
+        int* status) {
```

Comment:
All of these style changes also seem a bit gratuitous.  I usually prefer indenting to the opening `(` unless the function name is REALLY long, so you want to avoid cramming things into the right margin.  But that doesn't seem to be the case here.   

So again, I'd probably like some of the CCL coders to weigh in on this if they care and pick something consistent for the style of these function names.

Reply:
Happy to change it back. It was a huge hodge podge of spaces and tabs before which is why I went for it. I don't care how it actually looks. ;P

Reply:
I changed these back for the most part. Some fixes of tabs and spaces remain.